content/posts/2021/10/libplanet-0.11-0.19/index.eng.md

```diff
@@ -0,0 +1,251 @@
+
+The protocol does not require miners to have any priority for transactions
+to include in a block, if these staisfy
+[`IBlockPolicy<T>.ValidateNextBlockTx()` predicate
+method][IBlockPolicy<T>.ValidateNextBlockTx] at least.[^3]  However, if there
+are too many transactions to be included in a block, a miner may need to
+prioritize them so that they are chunked into multiple blocks.
+
+Since version 0.17.0, instead of Libplanet arbitrarily prioritizing them,
+miners can configure the priority for transactions to include in a block.
+API-wise, [`BlockChain<T>.MineBlock()` method][BlockChain<T>.MineBlock] has the
+`txPriority` option which takes an [`IComparer<Transaction<T>>`][IComparer<T>].
+
+[^3]: Of course, there is an order between transactions signed by a signer.
+      These are ordered by [`Transaction<T>.Nonce`][Transaction<T>.Nonce].
+
+[IBlockPolicy<T>.ValidateNextBlockTx]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Policies.IBlockPolicy-1.html#Libplanet_Blockchain_Policies_IBlockPolicy_1_ValidateNextBlockTx_Libplanet_Blockchain_BlockChain__0__Libplanet_Tx_Transaction__0__
+[Transaction<T>.Nonce]: https://docs.libplanet.io/0.19.0/api/Libplanet.Tx.Transaction-1.html#Libplanet_Tx_Transaction_1_Nonce
+[BlockChain<T>.MineBlock]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.BlockChain-1.html#Libplanet_Blockchain_BlockChain_1_MineBlock_Libplanet_Crypto_PrivateKey_DateTimeOffset_System_Boolean_System_Int32_System_Int32_IComparer_Libplanet_Tx_Transaction__0___CancellationToken_
+[IComparer<T>]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icomparer-1
+
+
+Blocks Signed By Miners
+-----------------------
+
+*Added in version [0.18.0].*
+
+Since the protocol version 2, i.e., Libplanet 0.18.0, every block is signed with
+the private key of its miner, and it includes the public key besides the addres
+of its miner.
+
+API-wise, block signatures are represented as
+[`Block<T>.Signature` property][Block<T>.Signature], and miners' public keys
+are represented as [`Block<T>.PublicKey` property][Block<T>.PublicKey].
+Blocks with the protocol versions 1 and 0 fill both properties with `null`.
+
+[Block<T>.Signature]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_Signature
+[Block<T>.PublicKey]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_PublicKey
+
+
+Multi-Threaded Mining
+---------------------
+
+*Added in version [0.19.0].*
+
+Since Libplanet 0.19.0, each block is mined in parallel, depending on
+the number of CPU cores.  The option to configure the number of processes
+(workers) will be added in the future.
+
+
+Many Others
+-----------
+
+Besides that, a lot of new features has been added, and it has significantly
+stabilized in many areas, solving problems that have arisen during months of
+the mainnet ops.  Please check the [entire changelog] of each release
+for details.
```

Comment:
```suggestion
In addition, lots of new features have been added, and it has significantly
stabilized the network in many areas, solving problems that have arisen during the past few months of
the mainnet ops.  Please check the [entire changelog] of each release
for details.
```

Reply:
Do we need the word 'entire'?

Reply:
I've added the word as it's a summary of multiple releases made over several months.